homeassistant/components/watchyourlan/sensor.py

```diff
@@ -0,0 +1,155 @@
+"""Support for the WatchYourLAN service."""
+
+from typing import Any
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import WatchYourLANUpdateCoordinator
+
+# Define entity descriptions for each sensor type
+ENTITY_DESCRIPTIONS = [
+    SensorEntityDescription(
+        key="online_status",
+        name="Online Status",
+        icon="mdi:lan-connect",
+    ),
+    SensorEntityDescription(
+        key="ip_address",
+        name="IP Address",
+        icon="mdi:ip-network",
+    ),
+    SensorEntityDescription(
+        key="mac_address",
+        name="MAC Address",
+        icon="mdi:lan",
+    ),
+    SensorEntityDescription(
+        key="iface",
+        name="Network Interface",
+        icon="mdi:ethernet",
+    ),
+]
+
+
+# Utility function for setting up device info consistently
+def build_device_info(device: dict[str, Any], domain: str) -> DeviceInfo:
+    """Build device info helper function."""
+    return DeviceInfo(
+        identifiers={(domain, device.get("ID", "Unknown"))},
+        name=device.get("Name") or f"WatchYourLAN {device.get('ID', 'Unknown')}",
+        manufacturer=device.get("Hw", "Unknown Manufacturer"),
+        model="WatchYourLAN Device",
+        sw_version=device.get("Last_Seen", "Unknown"),
```

Comment:
Is last seen the version?

Reply:
Not really, but I wasn't sure what else to use for this. There really is no "version". Would it be best to just leave sw_version unset?

Reply:
It's not required, so leave it unset if you don't have it